### PR TITLE
Decouple test from the default resolver for data_adapter

### PR DESCRIPTION
**Pull request by Gaurav0**
For #15058

### CODE REVIEW

File: packages/ember-extension-support/tests/data_adapter_test.js [hunk 1]
```diff
@@ -8,223 +8,233 @@ import {
...
-      App.register('data-adapter:main', DataAdapter);
-    });
   },
-  teardown() {
-    run(() => {
-      adapter.destroy();
-      App.destroy();
+  init() {
+    this._super(...arguments);
+    let self = this;
```
Comment (mixonic) [inline packages/ember-extension-support/tests/data_adapter_test.js]: I don't believe `self` is used here

File: packages/ember-extension-support/tests/data_adapter_test.js [hunk 2]
```diff
@@ -8,223 +8,233 @@ import {
...
-    deepEqual(postType.columns, [{ name: 'title', desc: 'Title' }], 'Correctly sets the columns');
-  }
 
-  run(adapter, 'watchModelTypes', modelTypesAdded);
-});
+moduleFor('Data Adapter', class extends ApplicationTestCase {
 
-QUnit.test('getRecords gets a model name as second argument', function() {
-  App.Post = Model.extend();
+  ['@test Model types added with DefaultResolver']() {
```
Comment (mixonic) [inline packages/ember-extension-support/tests/data_adapter_test.js]: So this test is specifically mentioning `DefaultResolver`. By migrating to the `ApplicationTestCase` and using `add` etc, it is no longer using the default resolver. Instead it now uses the TestResolver.

Either the test description is wrong and the data adapter doesn't require the DefaultResolver, or this test should not be ported. I''m not sure which.

File: packages/ember-extension-support/tests/data_adapter_test.js [hunk 3]
```diff
@@ -8,223 +8,233 @@ import {
...
-    });
   },
-  teardown() {
-    run(() => {
-      adapter.destroy();
-      App.destroy();
+  init() {
+    this._super(...arguments);
+    let self = this;
+    this.set('containerDebugAdapter', {
```
Comment (mixonic) [inline packages/ember-extension-support/tests/data_adapter_test.js]: I don't think this property is used?

File: packages/ember-extension-support/tests/data_adapter_test.js [hunk 4]
```diff
@@ -8,223 +8,233 @@ import {
 import { Object as EmberObject, A as emberA } from 'ember-runtime';
 import EmberDataAdapter from '../data_adapter';
 import { Application as EmberApplication, Resolver as DefaultResolver } from 'ember-application';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
+import { ModuleBasedResolver } from 'internal-test-helpers/test-resolver';
```
Comment (mixonic) [inline packages/ember-extension-support/tests/data_adapter_test.js]: This import is not used.

**Review by mixonic — CHANGES_REQUESTED** [review]
I believe that first test can just have its description changed. A read of https://github.com/emberjs/ember.js/blob/master/packages/ember-extension-support/lib/data_adapter.js doesn't make me believe these tests are specific to the default resolver, nor do these tests.

Some small cleanup remains. Thanks @Gaurav0!

File: packages/ember-extension-support/tests/data_adapter_test.js (hunk 3 above)
Comment (Gaurav0) [inline packages/ember-extension-support/tests/data_adapter_test.js]: Tests fail without a mock containerDebugAdapter.

File: packages/ember-extension-support/tests/data_adapter_test.js (hunk 1 above)
Comment (Gaurav0) [inline packages/ember-extension-support/tests/data_adapter_test.js]: Fixed.

File: packages/ember-extension-support/tests/data_adapter_test.js (hunk 4 above)
Comment (Gaurav0) [inline packages/ember-extension-support/tests/data_adapter_test.js]: Fixed.

File: packages/ember-extension-support/tests/data_adapter_test.js (hunk 2 above)
Comment (Gaurav0) [inline packages/ember-extension-support/tests/data_adapter_test.js]: Removing DefaultResolver from description

File: packages/ember-extension-support/tests/data_adapter_test.js (hunk 3 above)
Comment (mixonic) [inline packages/ember-extension-support/tests/data_adapter_test.js]: I misread this the first time, looks good 👍

**Review by mixonic — APPROVED** [review]

**Comment by mixonic** [comment]
Thank you @Gaurav0!

**Pull request merged**